src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -286,7 +286,7 @@ public function getName(/* $triggerDeprecation = true */)
         }
 
         if (null === $this->name) {
-            $this->name = preg_replace('/[^a-zA-Z0-9_]+/', '', basename($this->getProjectDir()));
+            $this->name = preg_replace('/[^a-zA-Z0-9_]+/', '', basename($this->rootDir));
```

Comment:
wasn't the point deprecating `$this->rootDir` ?

Reply:
Yes but since we are using it to compute `$this->name` which is deprecated too I don't think it's a problem to use it.

Reply:
deprecated feature using a deprecated feature (without triggering); that's OK and BC safe.